ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java

```diff
@@ -20,26 +20,66 @@
  * A message info class that contains basic info about a message
  */
 public class MessageInfo {
+
+  public static final short ACCOUNTID_CONTAINERID_DEFAULT_VALUE = -1;
+
   private final StoreKey key;
   private final long size;
-  private final long expirationTimeInMs;
-  private final boolean isDeleted;
-  private final Long crc;
+  private long expirationTimeInMs = Utils.Infinite_Time;
+  private boolean isDeleted = false;
+  private Long crc = null;
+  private short accountId;
+  private short containerId;
+  private long operationTimeMs;
 
-  public MessageInfo(StoreKey key, long size, long expirationTimeInMs) {
-    this(key, size, false, expirationTimeInMs);
-  }
+  public static class Builder {
```

Comment:
why do we need a builder for this? IMO, Builders are useful when there are a lot of mix and match options. In this case, you can either have a `MessageInfo` for a PUT or DELETE and at best, each can have its own constructor. Builders are also more useful when you have public APIs that users use - in this case, `MessageInfo` is internal. Builder looks like a bit of an overkill.